src/v0/destinations/kafka/transform.js

```diff
@@ -52,9 +53,10 @@ const batch = (destEvents) => {
       metadata: events.map((event) => event.metadata),
       destination: events[0].destination,
     };
-    respList.push(response);
+    respList.push(
+      getSuccessRespEvents(response.batchedRequest, response.metadata, response.destination, true),
```

Comment:
[Q] Should we also check for failed events and handle that as well ?

Reply:
Added handling for erroneous events.